src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -530,58 +537,254 @@ impl LogicalAggBuilder {
+                    inputs: inputs.clone(),
+                    distinct,
+                    order_by_fields: order_by_fields.clone(),
+                    filter: filter.clone(),
+                });
+                let sum_expr = ExprImpl::from(InputRef::new(
+                    self.group_key.len() + self.agg_calls.len() - 1,
+                    sum_return_type,
+                ))
+                .cast_implicit(return_type.clone())
+                .unwrap();
+
+                // then, we compute count
+                let count_return_type =
+                    AggCall::infer_return_type(&AggKind::Count, &[input.return_type()]).unwrap();
+
+                self.agg_calls.push(PlanAggCall {
+                    agg_kind: AggKind::Count,
+                    return_type: count_return_type.clone(),
+                    inputs,
+                    distinct,
+                    order_by_fields,
+                    filter,
+                });
 
-            let right = InputRef::new(
-                self.group_key.len() + self.agg_calls.len() - 1,
-                right_return_type,
-            );
+                let count_expr = ExprImpl::from(InputRef::new(
+                    self.group_key.len() + self.agg_calls.len() - 1,
+                    count_return_type,
+                ));
+
+                // we start with variance
+                let mut target_expr = ExprImpl::from(
+                    FunctionCall::new(
+                        ExprType::Divide,
+                        vec![
+                            ExprImpl::from(
+                                FunctionCall::new(
+                                    ExprType::Subtract,
+                                    vec![
+                                        squared_sum_expr,
+                                        ExprImpl::from(
+                                            FunctionCall::new(
+                                                ExprType::Divide,
+                                                vec![
+                                                    ExprImpl::from(
+                                                        FunctionCall::new(
+                                                            ExprType::Multiply,
+                                                            vec![sum_expr.clone(), sum_expr],
+                                                        )
+                                                        .unwrap(),
+                                                    ),
+                                                    count_expr.clone(),
+                                                ],
+                                            )
+                                            .unwrap(),
+                                        ),
+                                    ],
+                                )
+                                .unwrap(),
+                            ),
```

Comment:
Maybe we can add some intermediate variables and split the huge expression into pieces. :hot_face:

Reply:
fixed